.github/workflows/tests_and_reports.yml

```diff
@@ -126,59 +123,75 @@ jobs:
     if: ${{ always() }}
     runs-on: ubuntu-latest
+    env:
+      VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+      VERCEL_PROJECT_ID: ${{ secrets.VERCEL_META_SITE_PROJECT_ID }}
+      VERCEL_TOKEN: ${{secrets.VERCEL_META_SITE_DEPLOY_TOKEN }}
+      VERCEL_PREVIEW_URL: meta-site-${{ github.event.number }}-ldaf.vercel.app
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Create directory to collect reports
-        run: mkdir pr-reports
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+
+      - name: Install Vercel CLI
+        run: npm i -g vercel@latest
+
+      - name: Setup Vercel to build and deploy
+        run: vercel pull --yes --cwd .meta --environment=${{ fromJSON(env.IS_MAIN) && 'production' || 'preview' }} --token=${{env.VERCEL_TOKEN}}
 
       - name: Download unit test report artifact
         uses: actions/download-artifact@v3
         with:
           name: ${{ github.event.after }}-${{ env.UNIT_PATH }}
-          path: pr-reports/${{ env.UNIT_PATH }}
+          path: .meta/${{ env.UNIT_PATH }}
 
       - name: Download end-to-end test report artifact
         uses: actions/download-artifact@v3
         with:
           name: ${{ github.event.after }}-${{ env.E2E_PATH }}
-          path: pr-reports/${{ env.E2E_PATH }}
+          path: .meta/${{ env.E2E_PATH }}
 
       - name: Download bundle visualizer artifact
         uses: actions/download-artifact@v3
         with:
           name: ${{ github.event.after }}-${{ env.BUNDLE_VISUALIZER_PATH }}
-          path: pr-reports/${{ env.BUNDLE_VISUALIZER_PATH }}
+          path: .meta/${{ env.BUNDLE_VISUALIZER_PATH }}
 
-      - name: Set GitHub Pages preview URL for reports (for commenting)
-        if: github.ref != 'refs/heads/main'
-        run: echo "PREVIEW_URL=https://${{ github.repository_owner }}.github.io/${{ github.event.repository.name }}/${{ env.PR_PATH}}" >> $GITHUB_ENV
+      - name: Build for Vercel
+        run: vercel build --cwd .meta --token=${{env.VERCEL_TOKEN}} ${{ fromJSON(env.IS_MAIN) && '--prod'}}
 
-      - name: Deploy reports to GitHub Pages preview URL
-        # https://github.com/marketplace/actions/github-pages-action
-        uses: peaceiris/actions-gh-pages@v3
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_branch: gh-pages
-          publish_dir: ./pr-reports
-          destination_dir: ${{ github.ref == 'refs/heads/main' && env.MAIN_PATH || env.PR_PATH }}
+      - name: Deploy to Vercel preview
+        if: ${{ !fromJSON(env.IS_MAIN) }}
+        # Guide: https://vercel.com/guides/how-to-alias-a-preview-deployment-using-the-cli
+        run: |
+          url="$(vercel deploy --cwd .meta --prebuilt --token=${{env.VERCEL_TOKEN}})"
+          vercel alias --cwd .meta --scope ldaf --token=${{env.VERCEL_TOKEN}} set "$url" ${{env.VERCEL_PREVIEW_URL}}
```

Comment:
Unfortunately there's no way to set the URL as part of the `deploy` command, so we need to run `alias` after the fact to set it to a more predictable URL (I chose to use the PR number, since transforming the branch name into a slug would have been a little more complicated). If it would be useful, we could also set up an alias using the commit hash.